lib/rules/dot-notation.js

```diff
@@ -87,36 +87,38 @@ module.exports = {
                     data: {
                         key: formattedValue
                     },
-                    *fix(fixer) {
+                    fix(fixer) {
+                        const fixes = [];
                         const leftBracket = sourceCode.getTokenAfter(node.object, astUtils.isOpeningBracketToken);
                         const rightBracket = sourceCode.getLastToken(node);
                         const nextToken = sourceCode.getTokenAfter(node);
 
                         // Don't perform any fixes if there are comments inside the brackets.
                         if (sourceCode.commentsExistBetween(leftBracket, rightBracket)) {
-                            return; // eslint-disable-line eslint-plugin/fixer-return -- false positive
+                            return null;
```

Comment:
Isn't this a bug that should be fixed in `eslint-plugin-eslint-plugin`?

Reply:
created an issue https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/issues/117 to track it.

Reply:
reverted the changes; let's make another PR when it is fixed in the plugin.